Style of the sidebar for icon download

aminalhazwani opened the pull request:
I styled the sidebar following the design of the preferences sidebar of activity stream. I manually copied some of the styles (having a checkbox or radio button component would have been really helpful 😅). The CSS needs some cleaning/refactoring but I think we can do it before merging to production.

The naming of the options is temporary. In the meantime I placed 'Light' that corresponds to Grey 10 with 80% alpha channel and 'Dark' that corresponds to Grey 90 at 80%.

(I couldn't manage to run the website locally, if I run `npm run serve` I get a 404 page)

Commit:
[style] sidebar for icon download

Commit:
[add] opt-in toggle for mobile icons source files

Commit:
[minor] slowly adding photon color vars

Commit:
[minor] switch height and alignment

Commit:
[fix] switch active toggle

Commit:
[fix] icon details footer

bwinton commented on html/viewer/index.html:
```diff
@@ -5,35 +5,59 @@
-        <input type="checkbox" id="size-32"><label for="size-32">32x32</label><br>
-        <input type="checkbox" id="size-64"><label for="size-64">64x64</label><br>
-        <input type="checkbox" id="size-other"><label for="size-other">Other</label><br>
+  <div id="icon-details" class="show">
```
We shouldn't show the details on pageload.  There's no content in it, it's kinda confusing, and it hides some icons.

aminalhazwani replied:
Oh yeah, sure, I forgot to remove the class 😅

bwinton commented on html/viewer/css/app.css:
```diff
@@ -1,115 +1,333 @@
+  z-index: 1;
+  bottom: 0;
+  display: flex;
+  justify-content: flex-end;
```
The Cancel and Download buttons are too far over to the right.  This should probably be `center` or `space-around` (or maybe `space-between`)…

aminalhazwani replied:
I followed the spec of activity stream preferences sidebar

bwinton commented on html/viewer/css/app.css:
```diff
@@ -1,115 +1,333 @@
+  width: 16px;
+  height: 16px;
+  offset-inline-start: 0;
+  background: url("../images/check.svg") no-repeat center center;
```
I don't think we should have the `check.svg` as a background for the `:not(:checked)` case…

aminalhazwani replied:
Yes! I originally pushed in `debug` mode. It should now be fixed with e2fa817

bwinton commented on html/viewer/css/app.css:
```diff
@@ -1,115 +1,333 @@
+  content: '';
+  width: 12px;
+  height: 12px;
+  background-color: var(--blue-50);
```
Ditto here for the `background-color` of the `:not(:checked)` case…

aminalhazwani replied:
db04a04, I also added the `name` attribute to all radios and made `context-fill` the default one

bwinton commented on html/viewer/css/app.css:
```diff
@@ -1,115 +1,333 @@
+}
+
+#icon-details footer {
+  position: absolute;
```
On my iPhone 6s+ this hides the bottom of the details content.  Could we make this non-absolute, and add  `display:flex` to the `#icon-details` for layout instead?

aminalhazwani replied:
The website is not really responsive. When we implement the new styles we could also make it work beautifully no matter the screen size. I anyway changed the layout to use flex 89e46dc

bwinton commented on html/viewer/css/app.css:
```diff
@@ -1,115 +1,333 @@
 #icon-details {
   background: white;
-  height: calc(100vh - 70px);
+  height: 100vh;
```
I'm a little concerned that this will look not-so-great when we load the SharedBanner…

aminalhazwani replied:
Yeah, see 09637ee

bwinton commented on html/viewer/index.html:
```diff
@@ -5,35 +5,59 @@
-        </select>
+    </main>
+    <footer>
+      <div id="cancel">
```
The `cancel` button doesn't seem to do anything…  Did you need me to hook that up for you?

aminalhazwani replied:
ahhh I now see the close button up there. if we have the close button this is not necessary, I removed the cancel 1938930

bwinton commented on html/viewer/index.html:
```diff
@@ -49,6 +73,5 @@
   <script src="js/icon-styles.js"></script>
   <script src="js/icon-list.js"></script>
   <script src="js/app.js"></script>
-  <script src="https://firefoxux.github.io/SharedBanner/fxux-banner.min.js"></script>
```
ಠ_ಠ  Why no shared banner?

nt1m replied:
I think it looks a bit odd tbh, and not photon enough.

aminalhazwani replied:
Yeah, let's leave it for now and let's target it when we align the whole website style. See 09637ee

bwinton commented on html/viewer/css/app.css:
```diff
@@ -1,115 +1,333 @@
   text-transform: capitalize;
 }
+
+/** icon details checkbox **/
```
We should pull out the checkbox/radio button/slider css into its own file.

aminalhazwani replied:
See c378495

bwinton reviewed (changes requested):
There are definitely some things we want to fix before we land it, but I think it's on it's way!

Commit:
[edit] cancel/close button

Commit:
[edit] unchecked checkboxs by default

aminalhazwani commented on html/viewer/css/app.css:
```diff
@@ -1,115 +1,333 @@
   text-transform: capitalize;
 }
+
+/** icon details checkbox **/
```
Yes! I originally pushed in `debug` mode. It should now be fixed with e2fa817

Commit:
[edit] unchecked input radios by default

Commit:
[edit] hide include source file section

This section should be visible only if iOS and/or Android checkboxes are selected. By default if you download icons for one of these two platforms you will get only the PNGs.

nt1m commented on html/viewer/css/app.css:
```diff
@@ -174,18 +396,22 @@ body {
+
 .icon-display:active {
   background: #0a84ff;
   color: #f9f9fa;
```
This and above should probably use the variables as well.

aminalhazwani replied:
c96e46e

nt1m commented on html/viewer/index.html:
```diff
@@ -5,35 +5,57 @@
+          <div class="title">Platform</div>
+          <div class="checkbox">
+            <input type="checkbox" id="desktop-web" checked><label for="desktop-web"><span>Desktop/Web</span></label><br>
+          </div class="checkbox">
```
Here and below: closing tag should not include the class attribute

aminalhazwani replied:
Good catch! 73f4103

Commit:
[fix] class in closing tag

Commit:
[fix] class in closing tag

Commit:
[edit] replace hex with photon color vars

Commit:
[add] firefox ux banner

Commit:
[add] inputs css partials

Commit:
[edit] icon details with display flex

Commit:
[fix] sidebar layout with flexbox

aminalhazwani commented:
@bwinton I should have addressed all requested changes! If you agree I would keep the responsive work and the photon style alignment in a separate PR. If we can bind the radio buttons to the fill attribute of the SVGs I then think we can merge with production. These are the requirements:

- if 'Contenxt Fill' is selected the SVG should have `fill="context-fill"`
- if 'Dark' is selected the SVG should have `fill="#0c0c0d" fill-opacity="0.8"`
- if 'Light' is selected the SVG shoud have `fill="#f9f9fa" fill-opacity="0.8"`

I will open a couple of issues for feature requests and to dos!
